### ISSUE TITLE
[GRIN-Wallet] Test of wallet leaves TOR zombie processes

### ISSUE THREAD

**Issue by marekyggdrasil** [Grinnode-live/2020-grin-bug-bash-challenge#36]
**Description**
Testing the verify payment proof method from the Owner API

**Prerequisites**

1. GRIN-Node
2. GRIN-Wallet

**Test procedure**

1. Run GRIN-Wallet in Owner API listener mode
2. Perform many TOR transactions
3. Save output from system task manager such as `top`

**Expected result:**

Check if you can make TOR processes hang in the background as it was reported here https://github.com/mimblewimble/grin-wallet/issues/510

Provide requests and response full payloads as well as wallet and node logs.

**Comment by mojitoo** [comment]
how can i Perform many TOR transactions ? by opening multiple terminal ?

**Comment by marekyggdrasil** [comment]
I see the problem. Let's modify it a bit.

1. Run wallet in owner API listener mode and check it's process ID
2. Perform TOR transaction using another wallet and pay attention if TOR process gets spawned by the wallet
3. When transaction is finalized expect TOR process to get killed.

Replicate for 1, 2, 3 transactions in parallel, if that is not sufficient we can open a new issue (harder) for a more elaborate test.

It's an open discussion what's the best way to check it. Feel free to leave your opinion, how you see it.

**Comment by mojitoo** [comment]
do you mean i need to perform a transaction from wallet 2 to wallet 1 running in owner api mode ?

**Comment by mojitoo** [comment]
### Prerequisites:

* Wallet 1 : grin12wktxlyfx62wx48ldn55katd8zm5d6qfa6mupt9r4uul8eqxagsqct3je5
```
grin-wallet listen
```

After running the wallet in listener mode and make sure it was reachable, i can see by running _top_ command that **one** tor process was spawned. After 3 consecutive transactions, wallet didn’t spawn another tor process but reused same one. after i turned off the listen mode, tor process was killed.

**Comment by marekyggdrasil** [comment]
I'm convinced. Looks good to me! Please just include your wallet version in the last comment by sending the output of

```
grin-wallet -V
```

**Comment by mojitoo** [comment]
- grin-wallet 4.0.0
- Darwin MacBook-Pro-de-Workstation.local 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
